### DISCUSSION TITLE
Prometheus

### DISCUSSION THREAD

**Discussion by haf** [hasura/graphql-engine#6415] [Question]
Is it possible to monitor/observe the OSS version of Hasura with e.g. Jaeger/OpenTelemetry or Prometheus?

**Comment by m-rgba** [comment]
It's definitley possible to roll your own observability similar to how Hasura Cloud has monitoring and APM integrations.

To start - you may need to tweak the HASURA_GRAPHQL_ENABLED_LOG_TYPES env variable - which has  support for all of your service + performance logging: https://hasura.io/docs/1.0/graphql/core/deployment/logging.html

Then you can use a tool like Logspout to pipe those to Logstash for processing and then to Prometheus.

There are a couple of good articles on more Kubernetes specific observability tactics over on: https://hasura.io/docs/1.0/graphql/core/guides/monitoring/index.html#guides-monitoring-frameworks

**Comment by haf** [comment]
@mm-io I'm sorry, but that's not a solution. First of all Logstash, it's not even on the map of a modern deployment anymore (30 second startup times anyone). Also, Logstash doesn't support rollup metrics, and even if it did, the type of metrics you'd expect from a prometheus monitored service, like histogram, latencies and so on, aren't easily parsable from logs.

The integrations mentioned in those articles aren't even using the core feature sets of the products they concern. It's all logs. This is about metrics.

I appreciate you taking the time to answer though, even if it's not _the_ answer.
